pkg/cvo/sync_worker.go

```diff
@@ -608,6 +610,28 @@ func (w *SyncWorker) apply(ctx context.Context, payloadUpdate *payload.Update, w
 		// perform an orderly roll out by payload order, using some parallelization
 		// but avoiding out of order creation so components have some base
 		graph.Parallelize(payload.ByNumberAndComponent)
+		precreateObjects = true
```

Comment:
the clusteroperator sync tries to sync the related objects in precreating mode.

for during initlization and upgrade? mode.. cvo will try to reset the relatedObjs to what's in release-image when cluster operators might be adding new ones in the same time.

Reply:
> for during initlization and upgrade? mode.. cvo will try to reset the relatedObjs to what's in release-image when cluster operators might be adding new ones in the same time.

Yeah, this code defers all conflicts to avoid fighting.  The ones in the payload are a good starting point.


> for during initlization and upgrade?

Yes, in the end it should do this for both.